app/src/main/AndroidManifest.xml

```diff
@@ -6,6 +6,7 @@
   <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
   <uses-permission android:name="android.permission.INTERNET" />
   <uses-permission android:name="android.permission.WAKE_LOCK" />
+  <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
```

Comment:
I think this works for certain Android versions, but the MAC address isn't available otherwise and discouraged from being used (among all other hardware IDs).